src/xcode/ENA/HealthCertificateToolkit/Sources/HealthCertificateToolkit/ProofCertificateDownload.swift

```diff
@@ -0,0 +1,107 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+
+public struct ProofCertificateDownload {
+
+    // MARK: - Init
+
+    public init() {}
+
+    // MARK: - Public
+
+    public func fetchProofCertificate(
+        for healthCertificates: [Base45],
+        with httpService: HTTPServiceProtocol = HTTPService(),
+        completion: @escaping (Result<CBORData?, ProofCertificateFetchingError>
+    ) -> Void) {
+
+        let certificateAccess = DigitalGreenCertificateAccess()
+
+        let eligibleCertificates =
+            healthCertificates.compactMap { (base45) -> CBORData? in
+                let result = certificateAccess.extractCBOR(from: base45)
+                switch result {
+                case .success(let healthCertificateCBORData):
+                    return healthCertificateCBORData
+                case .failure:
+                    return nil
+                }
+            }.filter {
+                switch certificateAccess.extractDigitalGreenCertificate(from: $0) {
+                case .success(let certificate):
+                    return certificate.vaccinationCertificates[0].isEligibleForProofCertificate
+                case .failure:
+                    return false
+                }
+            }
```

Comment:
Just a formatting suggestion as I've seen this style quite often and like it.

```suggestion
        let eligibleCertificates = healthCertificates
            .compactMap { (base45) -> CBORData? in
                let result = certificateAccess.extractCBOR(from: base45)
                switch result {
                case .success(let healthCertificateCBORData):
                    return healthCertificateCBORData
                case .failure:
                    return nil
                }
            }
            .filter {
                switch certificateAccess.extractDigitalGreenCertificate(from: $0) {
                case .success(let certificate):
                    return certificate.vaccinationCertificates[0].isEligibleForProofCertificate
                case .failure:
                    return false
                }
            }
```